### ISSUE TITLE
POC needed CVE-2020-4074 - PS 1.6 workaround ?

### ISSUE THREAD

**Issue by touchweb-vincent** [PrestaShop/PrestaShop#20055]
Hello,

A POC is claimed against CVE-2020-4074 from our partners. It would be very appreciated if you can share one.

Is there a workaround for PS 1.6 since it seems to impact this version ?

Thanks,

Vincent

https://nvd.nist.gov/vuln/detail/CVE-2020-4074#vulnCurrentDescriptionTitle

**Comment by matks** [comment]
Hi, the advisory https://nvd.nist.gov/vuln/detail/CVE-2020-4074 also indicates the commit which fixes the issue https://github.com/PrestaShop/PrestaShop/commit/30b6a7bdaca9cb940d3ce462906dbb062499fc30 in version 1.7.6.6 .

As you can see from the commit, the fix requires 2 new SQL tables to store session data.

We cannot share any more information on how to implement it because it would also provide clear insights about how to _exploit_ it on unprotected shops.

However I think that following the guidelines from https://github.com/PrestaShop/PrestaShop/commit/30b6a7bdaca9cb940d3ce462906dbb062499fc30 and adapting them to previous versions such as PS 1.6 of PrestaShop is not complex. The structure of the session has not been modified a lot from PS 1.6 to PS 1.7.

I would also seize the opportunity to highlight that vulnerability searches are nowadays targeting PS 1.7 as this is the maintained version. This vulnerability was found because it is common to PS 1.6 and PS 1.7 however if there are vulnerabilities that only affect PS 1.6, they will probably stay hidden as there is no active research on the PS 1.6 scope (because PS 1.6 is EOL). Upgrading to PS 1.7 is the only safe way to stay secure.

**Comment by touchweb-vincent** [comment]
They ask a POC to confirm metrics of the CVE.

We will not advice our customers to spend their money on this upgrade, since everybody will be advice to upgrade to 1.7.7 for the PHP 7.3 compliance in .. a few weeks ?

Anyway we understand your point of view on sharing POC against CVE. But it's harder to provide wises advices, especialy in this period where people focus on essential spend.

Could you confirm us that it is a "true" critical update to apply without delay ?

Thanks you.

**Comment by matks** [comment]
> They ask a POC to confirm metrics of the CVE.
> 
> We will not advice our customers to spend their money on this upgrade, since everybody will be advice to upgrade to 1.7.7 for the PHP 7.3 compliance in .. a few weeks ?
> 
> Anyway we understand your point of view on sharing POC against CVE. But it's harder to provide wises advices, especialy in this period where people focus on essential spend.
> 
> Could you confirm us that it is a "true" critical update to apply without delay ?

None of the vulnerabilities disclosed last week are "critical" according to us. If they were critical, upon reporting, we would have released  a patch version fixing it within 7 days, just like we did for [PS 1.7.6.4](https://github.com/PrestaShop/PrestaShop/security/advisories/GHSA-mhfc-6rhg-fxp3).

When we receive vulnerability reports,
- if is it critical we fix it immediately and release a patch version as soon as possible (it takes 1 to 7 days depending on the release complexity)
- if it is not critical, we fix them in incoming patch or minor version

Most of the vulnerabilities disclosed last week require to be able to login into the Back Office so they cannot be exploited on any shop you just browse.

However they should not be underestimated.

**Comment by touchweb-vincent** [comment]
Thanks you for this reply,

Have a great day,

Vincent

**Comment by AfricanJoker** [comment]
> > They ask a POC to confirm metrics of the CVE.
> > We will not advice our customers to spend their money on this upgrade, since everybody will be advice to upgrade to 1.7.7 for the PHP 7.3 compliance in .. a few weeks ?
> > Anyway we understand your point of view on sharing POC against CVE. But it's harder to provide wises advices, especialy in this period where people focus on essential spend.
> > Could you confirm us that it is a "true" critical update to apply without delay ?
> 
> None of the vulnerabilities disclosed last week are "critical" according to us. If they were critical, upon reporting, we would have released a patch version fixing it within 7 days, just like we did for [PS 1.7.6.4](https://github.com/PrestaShop/PrestaShop/security/advisories/GHSA-mhfc-6rhg-fxp3).
> 
> When we receive vulnerability reports,
> 
>     * if is it critical we fix it immediately and release a patch version as soon as possible (it takes 1 to 7 days depending on the release complexity)
> 
>     * if it is not critical, we fix them in incoming patch or minor version
> 
> 
> Most of the vulnerabilities disclosed last week require to be able to login into the Back Office so they cannot be exploited on any shop you just browse.
> 
> However they should not be underestimated.

Hello @matks 

First of all, **thank you** for your excellent work on prestashop !

Sorry to dig up this dated post, but i have a pretty simple question regarding this **CVE-2020-4074** security flaw that has a **CVSS Score** of **10/10**.

I followed your answer on the following post (which we should not be underestimated, but you say that this one is not critical)

**So my question is ;**
- **Is this one exploitable by the frontend** (By a client or a user who does not know the administration url, simply by the frontend access) ?
- **Or this one requires an access in administration** (Back office account in logistician, translator, salesman, etc.) to be exploited ?

I can of course delete this "public" post after your answer in order to avoid giving clues to security researchers.

Thank you in advance for your time to answer me when you have a few moments.

Relatives links to CVE-2020-4074:
https://github.com/PrestaShop/PrestaShop/security/advisories/GHSA-ccvh-jh5x-mpg4
https://www.cvedetails.com/cve/CVE-2020-4074/
https://github.com/PrestaShop/PrestaShop/commit/30b6a7bdaca9cb940d3ce462906dbb062499fc30

**Comment by matks** [comment]
Thank you for reporting this topic: I found a big issue 😛 . We initially reported [CVE-2020-4074](https://www.cvedetails.com/cve/CVE-2020-4074/) with a CVSS of 5.7. On cvedetails it appears with a score of 10/10 this is a mistake. This vulnerability is NOT critical unless there is information I am not aware of.

> Is this one exploitable by the frontend (By a client or a user who does not know the administration url, simply by the frontend access) ?

Answer is no. We patched session management in both frontend and backend, but an attacker cannot get access to backend simply by the frontend access.

> Or this one requires an access in administration (Back office account in logistician, translator, salesman, etc.) to be exploited ?

I cannot give you such details but as mentioned before: this vulnerability is NOT critical unless there is information I am not aware of.

I will see how to fix cvdetails.com
